src/mesh/index_derivs.cxx

```diff
@@ -3015,12 +2950,8 @@ const Field3D Mesh::indexFDDY(const Field3D &v, const Field3D &f, CELL_LOC outlo
   Field3D result(this);
   result.allocate(); // Make sure data allocated
 
-  // There are four cases, corresponding to whether or not f and v
-  // have yup, ydown fields.
-
-  // If vUseUpDown is true, field "v" has distinct yup and ydown fields which
-  // will be used to calculate a derivative along
-  // the magnetic field
+  // If *UseUpDown is true, field "*" has distinct yup and ydown fields which
+  // will be used to calculate a derivative along the magnetic field
   bool vUseUpDown = (v.hasYupYdown() && ((&v.yup() != &v) || (&v.ydown() != &v)));
```

Comment:
I know you've not touched this, but do we really need to check that `yup/ydown` are distinct from the field? Is `hasYupYdown` not sufficient here?

Reply:
Without this check I think we wouldn't be able to use 5-point stencils with `paralleltransform=identity`. I haven't checked, but without this check I guess behaviour might even be inconsistent for identity transform, depending on whether a field had been communicated (when it would use the yup/ydown branch) or not (when it would use the to/fromFieldAligned branch).

I guess we could have ParallelTransformIdentity do nothing for calcYUpDown instead of using mergeYupYdown, then hasYupYdown would return false and we'd keep the current behaviour but maybe in a simpler/clearer way? If we did this we could get rid of mergeYupYdown and all the checks for `&f.yup()!=&f`.